### PR TITLE
Feature: Copy button

### PR DESCRIPTION
**Pull request by michaelessiet**
This PR simply contains the addition of the copy button to the code blocks

### CODE REVIEW

File: src/components/Snippet.tsx
Comment (Brian-Pob) [inline src/components/Snippet.tsx:1]: What's the purpose of this change? I'm not familiar with `PropsWithChildren` or `JSXElement` from the solid-js library.

Comment (michaelessiet) [reply src/components/Snippet.tsx:1]: Propswithchildren has been deprecated that's why I removed it
